From 1343f3723fafe970469bbee467ded518a749f730 Mon Sep 17 00:00:00 2001
From: Ewan Milne <emilne@redhat.com>
Date: Wed, 20 Aug 2014 18:58:40 -0400
Subject: [scsi] Add CDB definition for COMPARE_AND_WRITE

Message-id: <1408561176-18133-2-git-send-email-emilne@redhat.com>
Patchwork-id: 88419
O-Subject: [RHEL7.1 PATCH e-stor 01/57] scsi: Add CDB definition for COMPARE_AND_WRITE
Bugzilla: 1132092
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Ewan D. Milne" <emilne@redhat.com>

upstream commit 1c68cc1626341665a8bd1d2c7dfffd7fc852a79c

Tested by Ewan D. Milne and Barry Donahue with a combination
of manual and automated testing.  No issues seen.

upstream commit 1c68cc1626341665a8bd1d2c7dfffd7fc852a79c
Author: Nicholas Bellinger <nab@daterainc.com>
Date:   Mon Aug 19 13:48:10 2013 -0700

    scsi: Add CDB definition for COMPARE_AND_WRITE

    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Cc: Hannes Reinecke <hare@suse.de>
    Cc: Martin Petersen <martin.petersen@oracle.com>
    Cc: Chris Mason <chris.mason@fusionio.com>
    Cc: James Bottomley <JBottomley@Parallels.com>
    Cc: Nicholas Bellinger <nab@linux-iscsi.org>
    Signed-off-by: Nicholas Bellinger <nab@daterainc.com>

Signed-off-by: Ewan D. Milne <emilne@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/scsi/scsi.h b/include/scsi/scsi.h
index d477bfb..66d42ed 100644
--- a/include/scsi/scsi.h
+++ b/include/scsi/scsi.h
@@ -144,6 +144,7 @@ enum scsi_timeouts {
 #define ACCESS_CONTROL_IN     0x86
 #define ACCESS_CONTROL_OUT    0x87
 #define READ_16               0x88
+#define COMPARE_AND_WRITE     0x89
 #define WRITE_16              0x8a
 #define READ_ATTRIBUTE        0x8c
 #define WRITE_ATTRIBUTE       0x8d
-- 
1.7.1